### DISCUSSION TITLE
Why do I always redirect to HTTPS when accessing HTTP websites? How can I not redirect to HTTPS

### DISCUSSION THREAD

**Discussion by xiaoshangmin** [php-webdriver/php-webdriver#1085] [Q&A]
use selenium/standalone-chrome:4.16.1-20231219
Why do I always redirect to HTTPS when accessing HTTP websites? How can I not redirect to HTTPS

**Comment by uuf6429** [comment]
That is something you have to answer yourself because there are many ways why that redirect happens:
- in your site's code: https://stackoverflow.com/a/5106355/314056
- in your webserver: https://stackoverflow.com/a/16201658/314056
- or the browser itself when instructed by your server: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security
- or the many ways it can be done between the server and browser (e.g. load balancers, firewalls etc)
- or permanently enabled on the browser side: https://blog.chromium.org/2023/08/towards-https-by-default.html
- or it even by on the client side with javascript or meta headers (eww :))

You need to debug this yourself. First check who is causing the redirect: you can use `curl`/`wget` or check the browser dev tools to see if the target url is returning a HTTP Status 3xx with a Location header.
If you do not see that, then check for HSTS header and if that is also missing then it's most probably browser-side.

In any case, that sort of thing is not something that this project controls or handles.
